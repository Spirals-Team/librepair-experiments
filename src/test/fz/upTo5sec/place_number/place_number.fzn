array [1..2] of int: X_INTRODUCED_8 = [1,-1];
var 1..8: X_INTRODUCED_0;
var 1..8: X_INTRODUCED_1;
var 1..8: X_INTRODUCED_2;
var 1..8: X_INTRODUCED_3;
var 1..8: X_INTRODUCED_4;
var 1..8: X_INTRODUCED_5;
var 1..8: X_INTRODUCED_6;
var 1..8: X_INTRODUCED_7;
var -7..7: X_INTRODUCED_9 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_10 ::var_is_introduced ;
var -7..7: X_INTRODUCED_11 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_12 ::var_is_introduced ;
var -7..7: X_INTRODUCED_13 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_14 ::var_is_introduced ;
var -7..7: X_INTRODUCED_15 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_16 ::var_is_introduced ;
var -7..7: X_INTRODUCED_17 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_18 ::var_is_introduced ;
var -7..7: X_INTRODUCED_19 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_20 ::var_is_introduced ;
var -7..7: X_INTRODUCED_21 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_22 ::var_is_introduced ;
var -7..7: X_INTRODUCED_23 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_24 ::var_is_introduced ;
var -7..7: X_INTRODUCED_25 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_26 ::var_is_introduced ;
var -7..7: X_INTRODUCED_27 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_28 ::var_is_introduced ;
var -7..7: X_INTRODUCED_29 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_30 ::var_is_introduced ;
var -7..7: X_INTRODUCED_31 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_32 ::var_is_introduced ;
var -7..7: X_INTRODUCED_33 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_34 ::var_is_introduced ;
var -7..7: X_INTRODUCED_35 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_36 ::var_is_introduced ;
var -7..7: X_INTRODUCED_37 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_38 ::var_is_introduced ;
var -7..7: X_INTRODUCED_39 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_40 ::var_is_introduced ;
var -7..7: X_INTRODUCED_41 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_42 ::var_is_introduced ;
var -7..7: X_INTRODUCED_43 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_44 ::var_is_introduced ;
var -7..7: X_INTRODUCED_45 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_46 ::var_is_introduced ;
var -7..7: X_INTRODUCED_47 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_48 ::var_is_introduced ;
var -7..7: X_INTRODUCED_49 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_50 ::var_is_introduced ;
var -7..7: X_INTRODUCED_51 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_52 ::var_is_introduced ;
var -7..7: X_INTRODUCED_53 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_54 ::var_is_introduced ;
var -7..7: X_INTRODUCED_55 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_56 ::var_is_introduced ;
var -7..7: X_INTRODUCED_57 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_58 ::var_is_introduced ;
var -7..7: X_INTRODUCED_59 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_60 ::var_is_introduced ;
var -7..7: X_INTRODUCED_61 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_62 ::var_is_introduced ;
var -7..7: X_INTRODUCED_63 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_64 ::var_is_introduced ;
var -7..7: X_INTRODUCED_65 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_66 ::var_is_introduced ;
var -7..7: X_INTRODUCED_67 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_68 ::var_is_introduced ;
var -7..7: X_INTRODUCED_69 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_70 ::var_is_introduced ;
var -7..7: X_INTRODUCED_71 ::var_is_introduced :: is_defined_var;
var 2..7: X_INTRODUCED_72 ::var_is_introduced ;
array [1..8] of var int: x:: output_array([1..8]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7];
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_7],-1);
constraint int_abs(X_INTRODUCED_9,X_INTRODUCED_10);
constraint int_abs(X_INTRODUCED_11,X_INTRODUCED_12);
constraint int_abs(X_INTRODUCED_13,X_INTRODUCED_14);
constraint int_abs(X_INTRODUCED_15,X_INTRODUCED_16);
constraint int_abs(X_INTRODUCED_17,X_INTRODUCED_18);
constraint int_abs(X_INTRODUCED_19,X_INTRODUCED_20);
constraint int_abs(X_INTRODUCED_21,X_INTRODUCED_22);
constraint int_abs(X_INTRODUCED_23,X_INTRODUCED_24);
constraint int_abs(X_INTRODUCED_25,X_INTRODUCED_26);
constraint int_abs(X_INTRODUCED_27,X_INTRODUCED_28);
constraint int_abs(X_INTRODUCED_29,X_INTRODUCED_30);
constraint int_abs(X_INTRODUCED_31,X_INTRODUCED_32);
constraint int_abs(X_INTRODUCED_33,X_INTRODUCED_34);
constraint int_abs(X_INTRODUCED_35,X_INTRODUCED_36);
constraint int_abs(X_INTRODUCED_37,X_INTRODUCED_38);
constraint int_abs(X_INTRODUCED_39,X_INTRODUCED_40);
constraint int_abs(X_INTRODUCED_41,X_INTRODUCED_42);
constraint int_abs(X_INTRODUCED_43,X_INTRODUCED_44);
constraint int_abs(X_INTRODUCED_45,X_INTRODUCED_46);
constraint int_abs(X_INTRODUCED_47,X_INTRODUCED_48);
constraint int_abs(X_INTRODUCED_49,X_INTRODUCED_50);
constraint int_abs(X_INTRODUCED_51,X_INTRODUCED_52);
constraint int_abs(X_INTRODUCED_53,X_INTRODUCED_54);
constraint int_abs(X_INTRODUCED_55,X_INTRODUCED_56);
constraint int_abs(X_INTRODUCED_57,X_INTRODUCED_58);
constraint int_abs(X_INTRODUCED_59,X_INTRODUCED_60);
constraint int_abs(X_INTRODUCED_61,X_INTRODUCED_62);
constraint int_abs(X_INTRODUCED_63,X_INTRODUCED_64);
constraint int_abs(X_INTRODUCED_65,X_INTRODUCED_66);
constraint int_abs(X_INTRODUCED_67,X_INTRODUCED_68);
constraint int_abs(X_INTRODUCED_69,X_INTRODUCED_70);
constraint int_abs(X_INTRODUCED_71,X_INTRODUCED_72);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_1],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_2],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_3],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_4],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_5],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_2],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_3],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_4],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_5],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_3],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_4],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_5],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_3,X_INTRODUCED_4],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_3,X_INTRODUCED_5],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_3,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_3,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_4,X_INTRODUCED_5],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_4,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_4,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_5,X_INTRODUCED_6],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_5,X_INTRODUCED_7],0);
constraint int_lin_ne(X_INTRODUCED_8,[X_INTRODUCED_6,X_INTRODUCED_7],0);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_9],0):: defines_var(X_INTRODUCED_9);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_2,X_INTRODUCED_11],0):: defines_var(X_INTRODUCED_11);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_3,X_INTRODUCED_13],0):: defines_var(X_INTRODUCED_13);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_0,X_INTRODUCED_15],0):: defines_var(X_INTRODUCED_15);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_17],0):: defines_var(X_INTRODUCED_17);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_4,X_INTRODUCED_19],0):: defines_var(X_INTRODUCED_19);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_5,X_INTRODUCED_21],0):: defines_var(X_INTRODUCED_21);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_1,X_INTRODUCED_23],0):: defines_var(X_INTRODUCED_23);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_25],0):: defines_var(X_INTRODUCED_25);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_5,X_INTRODUCED_27],0):: defines_var(X_INTRODUCED_27);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_6,X_INTRODUCED_29],0):: defines_var(X_INTRODUCED_29);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_0,X_INTRODUCED_31],0):: defines_var(X_INTRODUCED_31);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_2,X_INTRODUCED_33],0):: defines_var(X_INTRODUCED_33);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_5,X_INTRODUCED_35],0):: defines_var(X_INTRODUCED_35);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_6,X_INTRODUCED_37],0):: defines_var(X_INTRODUCED_37);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_1,X_INTRODUCED_39],0):: defines_var(X_INTRODUCED_39);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_2,X_INTRODUCED_41],0):: defines_var(X_INTRODUCED_41);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_43],0):: defines_var(X_INTRODUCED_43);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_7,X_INTRODUCED_45],0):: defines_var(X_INTRODUCED_45);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_1,X_INTRODUCED_47],0):: defines_var(X_INTRODUCED_47);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_2,X_INTRODUCED_49],0):: defines_var(X_INTRODUCED_49);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_3,X_INTRODUCED_51],0):: defines_var(X_INTRODUCED_51);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_4,X_INTRODUCED_53],0):: defines_var(X_INTRODUCED_53);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_55],0):: defines_var(X_INTRODUCED_55);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_7,X_INTRODUCED_57],0):: defines_var(X_INTRODUCED_57);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_6,X_INTRODUCED_2,X_INTRODUCED_59],0):: defines_var(X_INTRODUCED_59);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_6,X_INTRODUCED_3,X_INTRODUCED_61],0):: defines_var(X_INTRODUCED_61);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_6,X_INTRODUCED_5,X_INTRODUCED_63],0):: defines_var(X_INTRODUCED_63);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_65],0):: defines_var(X_INTRODUCED_65);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_7,X_INTRODUCED_4,X_INTRODUCED_67],0):: defines_var(X_INTRODUCED_67);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_7,X_INTRODUCED_5,X_INTRODUCED_69],0):: defines_var(X_INTRODUCED_69);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_7,X_INTRODUCED_6,X_INTRODUCED_71],0):: defines_var(X_INTRODUCED_71);
solve :: int_search(x,first_fail,indomain_min,complete) satisfy;
