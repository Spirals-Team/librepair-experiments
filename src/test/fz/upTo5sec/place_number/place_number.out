x = array1d(1..8, [2, 5, 8, 6, 3, 1, 4, 7]);
----------